go/vt/vtgate/resilient_srv_topo_server.go

```diff
@@ -108,7 +108,7 @@ type srvKeyspaceEntry struct {
 	keyspace string
 
 	// the mutex protects any access to this structure (read or write)
-	mutex sync.Mutex
+	mutex sync.RWMutex
 
 	insertionTime time.Time
```

Comment:
Test for value==nil only is not sufficient.
If the watch call succeeds first, it has a valid value. Later if the topo server goes down, we need a way to reconnect and return cached value before the reconnection succeeds.